src/python/pants/backend/python/tasks2/resolve_requirements.py

```diff
@@ -0,0 +1,152 @@
+# coding=utf-8
+# Copyright 2016 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+import shutil
+from pex.fetcher import Fetcher
+from pex.interpreter import PythonInterpreter
+from pex.pex import PEX
+from pex.pex_builder import PEXBuilder
+from pex.platforms import Platform
+from pex.resolver import resolve
+from twitter.common.collections import OrderedSet
+
+from pants.backend.python.python_setup import PythonRepos, PythonSetup
+from pants.backend.python.targets.python_requirement_library import PythonRequirementLibrary
+from pants.base.fingerprint_strategy import DefaultFingerprintHashingMixin
+from pants.base.fingerprint_strategy import TaskIdentityFingerprintStrategy
+from pants.invalidation.cache_manager import VersionedTargetSet
+from pants.task.task import Task
+
+
+class PythonRequirementFingerprintStrategy(DefaultFingerprintHashingMixin,
+                                           TaskIdentityFingerprintStrategy):
+
+  def compute_fingerprint(self, req_lib):
+    hash_elements_for_target = []
+    hash_elements_for_target.extend([req.cache_key() for req in req_lib.requirements])
+    if not hash_elements_for_target:
```

Comment:
Its too bad `PythonRequirementLibrary` allows 0 requirements since that case is of no practical use that I can tell, and it just complicates code like this here.